tests/test_sqlalchemy_data_layer.py

```diff
@@ -767,14 +803,41 @@ def test_get_list_with_simple_filter(client, register_routes, person, person_2):
         querystring = urlencode(
             {
                 "page[number]": 1,
-                "page[size]": 1,
+                "page[size]": 10,
                 "fields[person]": "name,birth_date",
                 "sort": "-name",
                 "filter[name]": "test",
             }
         )
         response = client.get("/persons" + "?" + querystring, content_type="application/vnd.api+json")
         assert response.status_code == 200
+        assert len(response.json['data']) == 1
+
+
+def test_get_list_with_simple_filter_relationship_custom_qs(session, client, register_routes, person, person_2,
+                                                            computer, computer_2):
+    computer.person = person
+    computer_2.person = person_2
+    session.commit()
+    with client:
+        querystring = urlencode(
+            {
+                "filter[computers.id]": f'{computer_2.id},{computer.id}',
+                "include": "computers"
+            }
+        )
+        response = client.get("/persons_qs" + "?" + querystring, content_type="application/vnd.api+json")
+        assert response.status_code == 200
+        assert len(response.json['data']) == 2
+
+        querystring = urlencode(
+            {
+                "filter[computers.id]": f'{computer_2.id},{computer.id}',
+                "include": "computers"
+            }
+        )
+        response = client.get("/persons" + "?" + querystring, content_type="application/vnd.api+json")
+        assert response.status_code == 500
```

Comment:
this should be moved to a separate test. tests should be atomic. you can move repeating parts to fixtures